src/main/java/meditracker/medication/MedicationManager.java

```diff
@@ -85,6 +85,97 @@ public static Medication getMedication(String name) throws MedicationNotFoundExc
         throw new MedicationNotFoundException();
     }
 
+    /**
+     * Gets the Medication object from the medications list.
+     * Uses the Medication quantity to retrieve medications from the list.
+     * Shows all medications that is equal to or less than specified quantity,
+     *
+     * @param quantity Quantity of the medication to retrieve in double
+     * @throws MedicationNotFoundException No Medication matching specified name found
+     */
+    public static void showMedicationsByQuantity(Double quantity) throws MedicationNotFoundException {
+        int medicationsFound = 0;
+
+        for (Medication medication : medications) {
+            Double medicationListQuantity = medication.getQuantity();
+            if (Double.compare(medicationListQuantity, quantity) < 0
+                    || Double.compare(medicationListQuantity, quantity) == 0) {
```

Comment:
Might be better to combine it
```suggestion
            if (Double.compare(medicationListQuantity, quantity) <= 0) {
```

Reply:
Resolved in [5d76c97](https://github.com/AY2324S2-CS2113T-T09-1/tp/pull/206/commits/5d76c97bae5c72c936d0e27ba39e817825056a32)